src/discord/commands/report.ts

```diff
@@ -0,0 +1,136 @@
+        const messageCollector = interaction!.channel!.createMessageCollector({ time: 120000 });
+
+        let selectedroomId: string;
+        let selectedsectionId: string;
+        let roomSelectMenu: MessageActionRow;
+        let issueMessage = '';
+
+        selectMenuCollector.on('collect', async (menuInteraction: SelectMenuInteraction) => {
+            if (menuInteraction.user.id === interaction.user.id) {
+                switch (menuInteraction.customId) {
+                    case 'roomSelectMenu': {
+                        selectedroomId = menuInteraction.values[0];
+
+                        roomSelectMenu = new MessageActionRow().addComponents(new MessageSelectMenu().setCustomId('roomSelectMenu').addOptions(await retrieveRooms(selectedroomId)));
+                        const sectionSelectMenu = new MessageActionRow().addComponents(await retrieveSections(selectedroomId));
+
+                        menuInteraction.update({ embeds: [mainEmbed], components: [roomSelectMenu, sectionSelectMenu] });
+                        break;
+                    }
+                    case 'sectionSelectMenu': {
+                        //ESLint disabled for next line as regex is correct at removing unicode characters. Removes hidden unicode U+200E character that invalidates ObjectId casting
+                        selectedsectionId = menuInteraction.values[0].replace(/[^\x00-\x7F]/g, ''); //eslint-disable-line
+
+                        const sectionSelectMenu = new MessageActionRow().addComponents(await retrieveSections(selectedroomId, selectedsectionId));
+                        const confirmButton = new MessageActionRow().addComponents(new MessageButton().setCustomId('confirmReport').setLabel('Confirm Report').setStyle('SUCCESS'));
+                        const messageEmbed = new MessageEmbed()
+                            .setColor('#64ff5c')
+                            .setTitle(`Current Issue Description`)
+                            .setDescription('Type a message in chat to submit a message with the report.\n\n***A message preview will be displayed here***');
+
+                        menuInteraction.update({ embeds: [messageEmbed], components: [roomSelectMenu, sectionSelectMenu, confirmButton] });
+                        break;
+                    }
+                    default:
+                        console.log('Select menu not found.');
+                }
+            } else {
+                menuInteraction.reply({ content: "This select menu isn't for you!", ephemeral: true });
+            }
+        });
+
+        buttonCollector.on('collect', async (buttonInteraction: ButtonInteraction) => {
+            if (buttonInteraction.user.id === interaction.user.id) {
+                messageCollector.stop();
+
+                const newIssue = await IssueModel.create({
+                    timestamp: DateTime.now().setZone('America/Toronto').toJSDate(),
+                    message: issueMessage,
```

Comment:
Should we check that issueMessage isn't empty? (from a quick read I believe it can remain empty - let me know if I'm wrong)